src/toil/jobStores/utils.py

```diff
@@ -109,3 +110,24 @@ def single_attempt():
             yield
 
         yield single_attempt()
+
+default_delays = (0, 1, 1, 4, 16, 64)
+default_timeout = 300
+
+
+def retry_http(delays=default_delays, timeout=default_timeout, predicate=None):
+    """
+    >>> i = 0
+    >>> for attempt in retry_http():
+    ...     with attempt:
+    ...         i += 1
+    ...         raise urllib2.HTTPError('http://www.test.com', '408', 'some message', {}, None)
+    Traceback (most recent call last):
+    ...
+    HTTPError: HTTP Error 408: some message
+    >>> i > 1
+    True
+    """
+    def retryable_http_error(e):
+        return isinstance(e, urllib2.HTTPError) and e.code in ('503', '408', '500')
+    return retry(delays=delays, timeout=timeout, predicate=predicate or retryable_http_error)
```

Comment:
… then you can eliminate the `or` down here. 

Why do you have to deviate from the convention? All other retry_xyz implementations work that way.